src/test/java/api/requests/RequestScheduledNoticesProcessingTests.java

```diff
@@ -0,0 +1,249 @@
+  public void beforeRequestExpirationNoticeShouldBeSentAndDeleted()
+    throws MalformedURLException,
+    InterruptedException,
+    TimeoutException,
+    ExecutionException {
+
+    JsonObject noticeConfiguration = new NoticeConfigurationBuilder()
+      .withTemplateId(templateId)
+      .withRequestExpirationEvent()
+      .withBeforeTiming(Period.days(5))
+      .sendInRealTime(true)
+      .create();
+    setupNoticePolicyWithRequestNotice(noticeConfiguration);
+
+    LocalDate requestExpiration = LocalDate.now(UTC).plusDays(4);
+    IndividualResource request = requestsFixture.place(new RequestBuilder().page()
+      .forItem(item)
+      .withRequesterId(requester.getId())
+      .withRequestDate(DateTime.now())
+      .withStatus(OPEN_NOT_YET_FILLED)
+      .withPickupServicePoint(pickupServicePoint)
+      .withRequestExpiration(requestExpiration));
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(scheduledNoticesClient::getAll, hasSize(1));
+
+    scheduledNoticeProcessingClient.runRequestNoticesProcessing();
+    List<JsonObject> notices = patronNoticesClient.getAll();
+
+    assertThat(notices, hasSize(1));
+    assertThat(notices.get(0), getTemplateContextMatcher(templateId, request));
+  }
+
+  @Test
+  public void beforeRequestExpirationRecurringNoticeShouldBeSentAndUpdated()
+    throws MalformedURLException,
+    InterruptedException,
+    TimeoutException,
+    ExecutionException {
+
+    JsonObject noticeConfiguration = new NoticeConfigurationBuilder()
+      .withTemplateId(templateId)
+      .withRequestExpirationEvent()
+      .withBeforeTiming(Period.days(3))
+      .recurring(Period.days(1))
+      .sendInRealTime(true)
+      .create();
+    setupNoticePolicyWithRequestNotice(noticeConfiguration);
+
+    LocalDate requestExpiration = LocalDate.now(UTC).plusDays(3);
+    IndividualResource request = requestsFixture.place(new RequestBuilder().page()
+      .forItem(item)
+      .withRequesterId(requester.getId())
+      .withRequestDate(DateTime.now())
+      .withStatus(OPEN_NOT_YET_FILLED)
+      .withPickupServicePoint(pickupServicePoint)
+      .withRequestExpiration(requestExpiration));
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(scheduledNoticesClient::getAll, hasSize(1));
+
+    DateTime nextRunTimeBeforeProcessing = DateTime.parse(scheduledNoticesClient.getAll()
+      .get(0).getString("nextRunTime"));
+
+
+    scheduledNoticeProcessingClient.runRequestNoticesProcessing();
+    List<JsonObject> notices = patronNoticesClient.getAll();
+
+    DateTime nextRunTimeAfterProcessing = DateTime.parse(scheduledNoticesClient.getAll()
+      .get(0).getString("nextRunTime"));
+
+    assertThat(notices, hasSize(1));
+    assertThat(nextRunTimeBeforeProcessing, is(nextRunTimeAfterProcessing.minusDays(1)));
+    assertThat(notices.get(0), getTemplateContextMatcher(templateId, request));
+  }
+
+  @Test
+  public void beforeHoldExpirationNoticeShouldBeSentAndDeleted()
```

Comment:
Under what circumstances should it be deleted, when it is first sent, because it is not recurring?

Reply:
Yes, because is not recurring. Extended method's name to reflect this.